test/metabase/models/field_values_test.clj

```diff
@@ -128,6 +128,24 @@
   (sync/sync-database! db)
   (find-values field-values-id))
 
+(deftest implicit-deduplication-test
+  (let [now (t/zoned-date-time)
+        then (t/plus now (t/millis 1))]
+    (mt/with-temp [Database    {database-id :id} {}
+                   Table       {table-id :id}    {:db_id database-id}
+                   Field       {field-id :id}     {:table_id table-id}
+                   FieldValues _                 {:field_id field-id :type :full :values ["a" "b"] :human_readable_values ["A" "B"] :updated_at now}
+                   FieldValues _                 {:field_id field-id :type :full :values ["c" "d"] :human_readable_values ["C" "D"] :updated_at then}]
+
+      (testing "When we have two FieldValues rows in the database, "
+        (is (= 2 (count (t2/select FieldValues :field_id field-id :type :full :hash_key nil))))
+        (testing "we always return the most recently inserted row"
+          (is (= ["C" "D"] (:human_readable_values (field-values/get-latest-full-field-values field-id))))
```

Comment:
This test could pass by coincidence if the DB's query implementation selects the first inserted row. A stronger test would check that the last row by `updated_at` is returned regardless of insertion order. Perhaps this would be sufficient: insert three field values rows, where the middle one has the latest updated_at date.

Reply:
Good call-out